test/server/auth-test.js

```diff
@@ -90,37 +93,49 @@ describe('Authentication', () => {
 
         after(() => basicServer.close());
 
-        it('Should authorize with correct credentials', done => {
+        it('Should authorize with correct credentials', () => {
             session.getAuthCredentials = () => {
                 return {
                     username: 'username',
                     password: 'password'
                 };
             };
 
-            request(proxy.openSession('http://127.0.0.1:1507/', session), (err, res, body) => {
-                expect(body).equal('Access granted');
-                expect(res.statusCode).equal(200);
+            const options = {
+                url:                     proxy.openSession('http://127.0.0.1:1507/', session),
+                resolveWithFullResponse: true
+            };
 
-                done();
-            });
+            return request(options)
+                .then((res) => {
+                    expect(res.statusCode).equal(200);
+                    expect(res.body).equal('Access granted');
+                });
         });
 
-        it('Should not authorize with incorrect credentials', done => {
+        it('Should not authorize with incorrect credentials', () => {
             session.getAuthCredentials = () => {
                 return {
                     username: 'username',
                     password: 'invalidPassword'
                 };
             };
 
-            request(proxy.openSession('http://127.0.0.1:1507/', session), (err, res, body) => {
-                expect(body).equal('Access denied');
-                expect(res.statusCode).equal(401);
-                // NOTE: prevent showing the native credentials window.
-                expect(res.headers['www-authenticate']).to.be.undefined;
-                done();
-            });
+            const options = {
+                url:                     proxy.openSession('http://127.0.0.1:1507/', session),
+                resolveWithFullResponse: true
+            };
+
+            return request(options)
+                .then(() => {
+                    expect.fail(0, 1, 'Request should raise an "401" error');
```

Comment:
Use the simpler notation:
`expect.fail('Request should raise an "401" error');`

See tests in https://github.com/chaijs/chai/blob/e354beeb17a5879d4fe55f0b0c4349f2ecbb7ac2/test/expect.js#L248